### ISSUE TITLE
Displacement and subdivision through Embree

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/LuxCore#9]
Embree supports both displacement and subdivision: https://embree.github.io/api.html
Would it be a good idea to make that usable in LuxCore?
Probably with dedicated shapes as wrappers.

During my online research I also stumbled about this issue: https://github.com/embree/embree/issues/163
Might be something to look out for if LuxCore uses dynamic scenes.

**Comment by Dade916** [comment]
We can not use use Embree displacement and subdivision because it doesn't export the result of subdivision/displacement, it only allows you to test a ray intersection. I need the tranformed mesh (i.e. end result) especially for GPUs.
The guys at Intel have asked me if we were interested to use Embree displacement/subdivision but I provided the above answer (so they confirmed it wasn't possible at the time). If noting has changed in the mean time, we can not.

Indeed, I could patch the Embree code to export the data like I did with BVH builder but it is something I would like to avoid.

P.S. we don't use Embree dynamic scene.
